manifests/web/base.pp

```diff
@@ -1,15 +1,21 @@
 define rgbank::web::base(
-  $db_name = undef,
-  $db_user = undef,
-  $db_password = undef,
-  $db_host = undef,
   $version,
   $source,
   $listen_port,
   $install_dir,
+  $db_name = undef,
+  $db_user = undef,
+  $db_password = undef,
+  $db_host = undef,
   $custom_wp_config = undef,
 ) {
-  include apache
+  include ::apache
+  include ::apache::mod::php
+  include ::mysql::client
+  class {'::mysql::bindings':
```

Comment:
Instead of managing more shared middleware from the app component, I'd actually like to move towards managing less, since managing shared system resources from app components is an anti-pattern. These should be in the role::appserver class, which is declared in the App Servers node group on the PE Master.  Add you app servers to that node group and you're golden.